composer/core/data_spec.py

```diff
@@ -48,8 +48,12 @@ def _num_microbatches_split_mapping(m, num_microbatches: int):
     for k, v in m.items():
         if isinstance(v, torch.Tensor):
             chunked[k] = _num_microbatches_split_tensor(v, num_microbatches)
-        if isinstance(v, (List, Tuple)):
+        elif isinstance(v, (List, Tuple)):
             chunked[k] = _num_microbatches_split_list(v, num_microbatches)
+        elif isinstance(v, (int, float, str, bool)):
+            chunked[k] = [v] * num_microbatches
```

Comment:
so we replicate the int/float/str/bool `num_microbatches` times? Why is that?

Reply:
added a comment! Requested feature from Daniel